File: app/src/main/java/com/sakuna63/tumbin/application/di/component/LoginComponent.java
package com.sakuna63.tumbin.application.di.component;

import com.sakuna63.tumbin.application.activity.LoginActivity;
import com.sakuna63.tumbin.application.di.module.ActivityModule;
import com.sakuna63.tumbin.application.di.module.LoginPresenterModule;
import com.sakuna63.tumbin.application.di.scope.ActivityScope;

import dagger.Component;

@ActivityScope
@Component(dependencies = ApplicationComponent.class,
        modules = {ActivityModule.class, LoginPresenterModule.class})
public interface LoginComponent extends ActivityComponent {
    void inject(LoginActivity activity);
}

File: app/src/main/java/com/sakuna63/tumbin/application/di/component/LoginComponent.kt
package com.sakuna63.tumbin.application.di.component

import com.sakuna63.tumbin.application.activity.LoginActivity
import com.sakuna63.tumbin.application.di.module.ActivityModule
import com.sakuna63.tumbin.application.di.module.LoginPresenterModule
import com.sakuna63.tumbin.application.di.scope.ActivityScope
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = arrayOf(ActivityModule::class, LoginPresenterModule::class))
interface LoginComponent : ActivityComponent {
    fun inject(activity: LoginActivity)
}
